src/garage/tf/algos/batch_polopt2.py

```diff
@@ -0,0 +1,281 @@
+"""Base class for batch sampling-based policy optimization methods."""
+
+import collections
+
+from dowel import logger, tabular
+import numpy as np
+
+from garage.misc import tensor_utils as np_tensor_utils
+from garage.np.algos import RLAlgorithm
+from garage.sampler import OnPolicyVectorizedSampler
+from garage.tf.misc import tensor_utils
+from garage.tf.samplers import BatchSampler
+
+
+class BatchPolopt2(RLAlgorithm):
+    """Base class for batch sampling-based policy optimization methods.
+
+    This includes various policy gradient methods like VPG, NPG, PPO, TRPO,
+    etc.
+
+    Args:
+        env_spec (garage.envs.EnvSpec): Environment specification.
+        policy (garage.tf.policies.base.Policy): Policy.
+        baseline (garage.tf.baselines.Baseline): The baseline.
+        scope (str): Scope for identifying the algorithm.
+            Must be specified if running multiple algorithms
+            simultaneously, each using different environments
+            and policies.
+        max_path_length (int): Maximum length of a single rollout.
+        discount (float): Discount.
+        gae_lambda (float): Lambda used for generalized advantage
+            estimation.
+        center_adv (bool): Whether to rescale the advantages
+            so that they have mean 0 and standard deviation 1.
+        positive_adv (bool): Whether to shift the advantages
+            so that they are always positive. When used in
+            conjunction with center_adv the advantages will be
+            standardized before shifting.
+        fixed_horizon (bool): Whether to fix horizon.
+        flatten_input (bool): Whether to flatten input along the observation
+            dimension. If True, for example, an observation with shape (2, 4)
+            will be flattened to 8.
+
+    """
+
+    def __init__(self,
+                 env_spec,
+                 policy,
+                 baseline,
+                 scope=None,
+                 max_path_length=500,
+                 discount=0.99,
+                 gae_lambda=1,
+                 center_adv=True,
+                 positive_adv=False,
+                 fixed_horizon=False,
+                 flatten_input=True):
+        self.env_spec = env_spec
```

Comment:
Please change these to _private

Reply:
Constructor argument `fixed_horizon` is not being used. Is it supposed to allow us to return either dense or flattened samples?